apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1299 @@
+	 * 'server' address structure.  The system will assign me an arbitrary
+	 * local port that is not in use.
+	 */
+	ret = connect(sockfd, (struct sockaddr *)&server, sizeof(struct sockaddr_in));
+	if (ret < 0) {
+		PRNT("ERROR: connect failed: %d errno: %d", ret, errno);
+		close(sockfd);
+		return WGET_SOCKET_CONNECT_ERR;
+	}
+	FUNC_EX;
+	return sockfd;
+}
+
+#ifdef CONFIG_NET_SECURITY_TLS
+static int ws_test_wget_base(void *arg)
+{
+	FUNC_EN;
+	int sockfd = -1;
+	int ret;
+	int buf_len, sndlen, len;
+	int encoding = CONTENT_LENGTH;
+	int state = HTTP_REQUEST_HEADER;
+	struct http_message_len_t mlen = {0,};
+	struct ws_test_wget_s ws;
+	struct http_client_request_t *param = (struct http_client_request_t *)arg;
+	struct http_client_response_t response = {0, };
+	int read_finish = false;
+	int send_len;
+
+	struct http_client_tls_t *client_tls = (struct http_client_tls_t *)malloc(sizeof(
+			struct http_client_tls_t));
+	int handshake_retry = WEBCLIENT_CONF_HANDSHAKE_RETRY;
+
+	if (client_tls == NULL) {
+		PRNT("memory not allocated");
+		return -1;
+	}
+
+	/* Initialize the state structure */
+	memset(&ws, 0, sizeof(struct ws_test_wget_s));
+	ws.buffer = param->buffer;
+	ws.buflen = param->buflen;
+
+	/* Parse the hostname (with optional port number) and filename from the URL */
+	ret = netlib_parsehttpurl(param->url, &ws.port, ws.hostname, WS_TEST_MAXHOSTNAME, ws.filename, WS_TEST_MAXFILENAME);
+	if (ret != 0) {
+		PRNT("ERROR: Malformed HTTP URL: %s", param->url);
+		free(client_tls);
+		return WGET_ERR;
+	}
+
+	PRNT("hostname='%s' filename='%s'", ws.hostname, ws.filename);
+
+	if (param->tls && ws_test_webclient_tls_init(client_tls, &param->ssl_config)) {
+		PRNT("Fail to client tls init");
+		goto errout_before_tlsinit;
+	}
+
+	/* Re-initialize portions of the state structure that could have
+	 * been left from the previous time through the loop and should not
+	 * persist with the new connection.
+	 */
+	ws.httpstatus = HTTPSTATUS_NONE;
+	ws.offset = 0;
+	ws.datend = 0;
+	ws.ndx = 0;
+
+	if ((sndlen = ws_test_wget_msg_construct(ws.buffer, param, &ws)) <= 0) {
+		PRNT("ERROR: construction message failed");
+		goto errout_before_tlsinit;
+	}
+
+	PRNT("sndlen:%d , param->buffer is:", sndlen);
+	PRNT("%s", param->buffer);
+
+retry:
+	if ((sockfd = ws_test_wget_socket_connect(&ws)) < 0) {
+		PRNT("ERROR: socket failed: %d", errno);
+		goto errout_before_tlsinit;
+	}
+
+	client_tls->client_fd = sockfd;
+	if (param->tls && (ret = ws_test_wget_tls_handshake(client_tls, ws.hostname))) {
+		if (handshake_retry-- > 0) {
+			if (ret == MBEDTLS_ERR_NET_SEND_FAILED ||
+					ret == MBEDTLS_ERR_NET_RECV_FAILED ||
+					ret == MBEDTLS_ERR_SSL_CONN_EOF) {
+				PRNT("Handshake again.... ");
+				mbedtls_net_free(&(client_tls->tls_client_fd));
+				mbedtls_ssl_free(&(client_tls->tls_ssl));
+				goto retry;
+			}
+		}
+		goto errout;
+	}
+
+	send_len = sndlen;
+	for (int i = 0; i <= g_keep_alive_count; i++) {
+		if (!param->encoding) {
```

Comment:
the depth of condition is too deep. how about to divide the code to separated function.

Reply:
Made a separate function to process chunk transfer

Reply:
It seems that my comment was not clear. my suggestion is like following.
I think it's more  readable to understand code. 
```
static int ws_test_send(...) // separate function
{
if (!param->encoding) {
  ws_test_send(...)
} else {
  ws_test_chunk_transfer(...)
}

static int ws_test_base(...)
{
  if (ws_test_connect != 0) {
    // error handling
  } 
  for ( ) {
    if (ws_test_send() != 0) {
      // error handling
    }
    if (ws_test_recv() != 0) {
      // error handling
    }
  }
}
```